### DISCUSSION TITLE
Setting Observable Map Name

### DISCUSSION THREAD

**Discussion by BrianHung** [mobxjs/mobx-state-tree#2011] [Q&A] [answered]
I'm using the [trace utility from mobx](https://mobx.js.org/understanding-reactivity.html) and am noticing that with `types.map(Node)`, the name of the map is `ObservableMap@185` even though in the constructor, it uses the subtype to construct a name: https://github.com/mobxjs/mobx-state-tree/blob/50e6df954da72fd2875bffb14bda0d59e134c792/packages/mobx-state-tree/src/types/complex-types/map.ts#L504-L506

Is there a way to fix it to get `ObservableMap<Node>` or something readable when using trace with MST?

**Comment by coolsoftwaretyler** [comment]
Hey @BrianHung - I don't know how to solve this right now, but if it's something you'd like fixed, we are starting to get a decent influx of contributors. 

Would you be down to either: 

1. Submit a PR with some tests, documents, and the fix
2. Submit a PR just with some failing tests, maybe a code snippet of what you'd like

Even if you just submit a failing test, I can try to find someone to work on it (or perhaps work on it myself).

**Reply by BrianHung** [reply, marked as answer]
I'll try for a PR but let me look into the MST codebase a bit more.

**Reply by coolsoftwaretyler** [reply]
I'm going to mark this as answered with the merge in https://github.com/mobxjs/mobx-state-tree/pull/2049, hopefully should be out in next minor.
